File: cpp/121-best-time-to-buy-and-sell/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int _min = prices[0];
        int answer = 0;

        for (int i=1; i<prices.size(); i++) {
            answer = max(answer, prices[i] - _min);
            _min = min(_min, prices[i]);  
        }

        return answer;
    }
};

File: python/121-best-time-to-buy-and-sell/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        result = 0
        lowest = 0
        
        for now in range(1, len(prices)):
            if prices[now] < prices[lowest]:
                lowest = now
            result = max(result, prices[now] - prices[lowest])
            
        return result